### PR TITLE
Wildcard cleanup

### PR DESCRIPTION
**Pull request by jberggg**
I wrote tests for the wildcard matching and realised that there are some more special cases which are not yet properly handled. Thus besides the tests I also introduced some changes for the matching.

You can run the matcher tests with `sbt "testOnly com.sandinh.paho.akka.TopicMatching"`.

### CODE REVIEW

File: src/main/scala/com/sandinh/paho/akka/MqttPubSub.scala [hunk 1]
```diff
@@ -18,38 +18,55 @@ object MqttPubSub {
...
-    val sub = subTopic.split("/")
-    val actual = actualTopic.split("/")
+
+    val levelSeparator = "/"
+    val multiLevelWildcard = "#"
+    val singleLevelWildcard = "+"
+
+    def prepareTopicsForMatching(actual: String, subscribed: String): (Array[String], Array[String]) = {
+
+      def trailingLevelSeparator(s: String): Boolean = s.substring(s.length-1) == levelSeparator
```
Comment (mwysokinski) [inline src/main/scala/com/sandinh/paho/akka/MqttPubSub.scala]: s.endsWith(levelSeparator)

File: src/main/scala/com/sandinh/paho/akka/MqttPubSub.scala [hunk 2]
```diff
@@ -18,38 +18,55 @@ object MqttPubSub {
...
+
+      def topicSplit(s: String) =
+        if(trailingLevelSeparator(s))
+          s.split(levelSeparator) :+ ""
+        else
+          s.split(levelSeparator)
```
Comment (mwysokinski) [inline src/main/scala/com/sandinh/paho/akka/MqttPubSub.scala]: what is happening with value of this if?

**Review by mwysokinski — APPROVED** [review]
LGTM

File: src/main/scala/com/sandinh/paho/akka/MqttPubSub.scala [hunk 3]
```diff
@@ -18,38 +18,54 @@ object MqttPubSub {
...
         logger.debug("matched single-level wildcard")
         true
       }
-      else if (subscribed == "#"){
+      else if (subscribed == multiLevelWildcard ){
         logger.debug("matched multi-level wildcard")
         true
       }
       else
         false
```
Comment (spoltier) [inline src/main/scala/com/sandinh/paho/akka/MqttPubSub.scala]: this if block seems like a good candidate for a match / case statement, unless you plan to remove the log statements.

**Review by spoltier — APPROVED** [review]

**Pull request merged**